WC_Layers/Source_Mesh_Reader.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 30 15:16:44 2024
+
+@author: Anupama Rajendra
+"""
+
+import h5py
+import numpy as np
+
+source_density_name = 'source_density'
+
+# Open the HDF5 files
+File_1 = h5py.File("source_mesh_1.h5m", 'r')
+File_2 = h5py.File("source_mesh_2.h5m", 'r')
+Files = [File_1, File_2]
+
+def extract_source_data(file_list):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
+    sd_list = []
+    for file in file_list:
+        # Extract data from the current file
+        tstt = file['tstt']
+        elements = tstt['elements']
+        tet_four = elements['Tet4']
+        tags = tet_four['tags']
+        sd = tags['source_density'][:]
+        sd_list.append(sd)
+    return sd_list
+    
+def arrange_source_strengths(sd_list, photon_groups):
+    '''
+    Writes individual mesh element strengths (separated by energy group) to density_list, 
+    then writes density_list to all_strengths_list for each set of source densities. 
+    '''
+    all_strengths_list = []
+    for density in sd_list:
+        density_list = []
+        # Calculate (individual mesh) strengths for each photon group
+        for group in range(photon_groups):
+            strengths = density[:,group]
+            density_list.append(strengths)
+        all_strengths_list.append(density_list)    

```

Comment:
I am not sure I understand the shape of these data structures, but looking at the operations, it seems like this could be one or two lines using numpy operations.  I recommend that all data be treated as numpy arrays as soon as possible

Reply:
If I read this correctly, you are looping over the data from each file (i.e. each shutdown time), each of which is size (num_mesh_elements, num_groups) and making a list that is (num_groups, num_mesh_elements).  That is, you are just transposing this data.  If it was a numpy array of size `(num_files, num_mesh_elements, num_groups)` this would be one line - something like:
```
np.transpose(sd_list, (0,2,1))
```